test/EndToEnd/GivenDotNetUsesMSBuild.cs

```diff
@@ -70,6 +70,25 @@ public void ItCanRunToolsInACSProj()
         }
 
         [Fact]
+        public void ItCanRunToolsThatPrefersTheCliRuntime()
+        {
+            var testInstance = TestAssets.Get("MSBuildTestApp")
+                                         .CreateInstance()
+                                         .WithSourceFiles()
+                                         .WithRestoreFiles();
+         
+            var testProjectDirectory = testInstance.Root;
+
+            new DotnetCommand()
+                .WithWorkingDirectory(testInstance.Root)
+                .ExecuteWithCapturedOutput("prefercliruntime")
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello I prefer the cli runtime World!");;
```

Comment:
I do see that we have the app target 1.1 and therefore prove the scenario today. However, If we ever do a 3.1 or equivalent then this test's result will become ambiguous. I'd feel better if:

a) the package targets a non-standard [e.g. a 1.1 preview] Microsoft.NetCore.App package
b) There were a `because` argument here that explains what's really going on... I'll forget the context :(